pkg/auth/oauth/registry/timeoutvalidator.go

```diff
@@ -16,30 +15,27 @@ import (
 	oauthclient "github.com/openshift/origin/pkg/oauth/generated/internalclientset/typed/oauth/internalversion"
 	oauthclientlister "github.com/openshift/origin/pkg/oauth/generated/listers/oauth/internalversion"
 	"github.com/openshift/origin/pkg/user/apis/user"
+	"github.com/openshift/origin/pkg/util/rankedset"
 )
 
 var errTimedout = errors.New("token timed out")
 
+// Implements rankedset.Item
 type tokenData struct {
 	token *oauth.OAuthAccessToken
 	seen  time.Time
 }
 
-func (a *tokenData) timeout() time.Time {
+func (a tokenData) timeout() time.Time {
```

Comment:
You probably want to use *tokenData to avoid copying the struct unnecessarily.

Reply:
I was doing that before, but when I added RankedSet the compiler complained, changing it to not be apointer made it work.

Reply:
Right because you will have to update everything to use `*tokenData` otherwise it will not implement the interface.